src/query/catalog/src/statistics/basic_statistics.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_storage::Datum;
+use storages_common_table_meta::meta::ColumnStatistics;
+
+// #[derive(Debug, Clone)]
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, Default)]
+/// Basic statistics information of a column
+pub struct BasicColumnStatistics {
+    /// Min value of the column
+    pub min: Option<Datum>,
+    /// Max value of the column
+    pub max: Option<Datum>,
+    // Number of Distinct Value
+    pub ndv: Option<u64>,
+    // Count of null values
+    pub null_count: u64,
+}
+
+impl From<ColumnStatistics> for BasicColumnStatistics {
+    fn from(value: ColumnStatistics) -> Self {
+        Self {
+            min: Datum::from_scalar(value.min),
+            max: Datum::from_scalar(value.max),
+            ndv: value.distinct_of_values,
+            null_count: value.null_count,
+        }
+    }
+}
+
+impl BasicColumnStatistics {
+    pub fn new_null() -> Self {
+        Self {
+            min: None,
+            max: None,
+            ndv: None,
+            null_count: 0,
+        }
+    }
+
+    pub fn merge(&mut self, other: BasicColumnStatistics) {
+        self.min = Datum::min(self.min.clone(), other.min);
+        self.max = Datum::max(self.max.clone(), other.max);
+        self.ndv = match (self.ndv, other.ndv) {
+            (Some(x), Some(y)) => Some(x + y),
```

Comment:
It's possible that the real ndv is less than the plus result?

Reply:
Yes, it depends on parquet's metadata, we do this to avoid losing information.